app/src/components/UnitForm/index.js

```diff
@@ -1,188 +1,353 @@
-import React, { useState } from "react";
+import React, { useState, useEffect } from "react";
+import { useNavigate } from "react-router-dom";
+import toast, { Toaster } from "react-hot-toast";
 import {
   TextField,
   Button,
   Typography,
   Paper,
   MenuItem,
   IconButton,
+  FormControlLabel,
+  Checkbox,
+  InputAdornment,
   Box,
 } from "@mui/material";
 import CloseIcon from "@mui/icons-material/Close";
 import FileBase from "react-file-base64";
 import axios from "axios";
 
-const UnitForm = ({ handleClose }) => {
+const UnitForm = ({ handleClose, addressDict }) => {
+  const currentAddresses = Object.keys(addressDict);
+  const navigate = useNavigate();
+
   const leaseOptions = [
     { value: 4, label: "4 months" },
     { value: 8, label: "8 months" },
     { value: 12, label: "12 months" },
   ];
 
+  const laundryOptions = ["Building", "Ensuite", "None"];
+  const typeOptions = ["Apartment", "House"];
+
   const defaultUnitValues = {
-    address: "301 Phillip St",
-    room: 3,
-    price: 500,
-    numBeds: 5,
-    numWashrooms: 5,
-    leaseDuration: 4,
-    floor: 3,
-    selectedImage: "",
+    address: "",
+    room_num: "",
+    rent_price: 500,
```

Comment:
Is there a point for having things like rent_price, num_beds pre-populated instead of being set to 0?

Reply:
I did this so that there's some default value on the form when opened, but if we don't want default value, I can just set them to be empty too.